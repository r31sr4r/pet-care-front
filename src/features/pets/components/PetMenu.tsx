import {
	Box,
	Divider,
	IconButton,
	Tooltip,
	Typography,
	Grid,
} from '@mui/material';
import VaccinesIcon from '@mui/icons-material/Vaccines';

type Props = {
	petId: string;
	handleClick: (petID: any) => void;
};

export const PetMenu = ({ petId, handleClick }: Props) => {
	return (
		<Box mb={2}>
			<Grid container spacing={2} direction="column">
				<Grid item xs={3}>
					<Tooltip
						title="Vacinas"
						placement="left-start"
						sx={{ mr: 0.3, ml: 2 }}
					>
						<IconButton
							edge="end"
							aria-label="vacinas"
							onClick={() => handleClick(petId)}
						>
							<VaccinesIcon fontSize="large" />
						</IconButton>
					</Tooltip>
					<Grid item xs={3}>
						<Typography
							variant="caption"
							gutterBottom
							sx={{ ml: 2.3 }}
						>
							Vacinas
						</Typography>
					</Grid>
				</Grid>
			</Grid>
			<Divider />
		</Box>
	);
};
