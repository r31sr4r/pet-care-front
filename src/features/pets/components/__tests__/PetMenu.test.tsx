import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { PetMenu } from '../PetMenu';

const Props = {
    petId: '78c15db6-8ce1-460b-9bd7-2dca5e9bfa9d',
    handleClick: () => {},
};

describe('PetMenu', () => {
    it('should render successfully', () => {
        const { asFragment } = render(<PetMenu {...Props} />, {
            wrapper: BrowserRouter,
        });
        expect(asFragment()).toMatchSnapshot();
    });
});