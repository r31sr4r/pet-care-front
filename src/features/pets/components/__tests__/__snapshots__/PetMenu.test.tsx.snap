// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetMenu should render successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1qm1lh"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1rl1og4-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      >
        <button
          aria-label="vacinas"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1jdufwn-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="VaccinesIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 5.5H8V4h.5c.55 0 1-.45 1-1s-.45-1-1-1h-3c-.55 0-1 .45-1 1s.45 1 1 1H6v1.5H3c-.55 0-1 .45-1 1s.45 1 1 1V15c0 1.1.9 2 2 2h1v4l2 1.5V17h1c1.1 0 2-.9 2-2V7.5c.55 0 1-.45 1-1s-.45-1-1-1zM9 9H7.25c-.41 0-.75.34-.75.75s.34.75.75.75H9V12H7.25c-.41 0-.75.34-.75.75s.34.75.75.75H9V15H5V7.5h4V9zm10.5 1.5V10c.55 0 1-.45 1-1s-.45-1-1-1h-5c-.55 0-1 .45-1 1s.45 1 1 1v.5c0 .5-1.5 1.16-1.5 3V20c0 1.1.9 2 2 2h4c1.1 0 2-.9 2-2v-6.5c0-1.84-1.5-2.5-1.5-3zm-3 0V10h1v.5c0 1.6 1.5 2 1.5 3v.5h-4v-.5c0-1 1.5-1.4 1.5-3zM15 20v-1.5h4V20h-4z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption MuiTypography-gutterBottom css-zl89d7-MuiTypography-root"
          >
            Vacinas
          </span>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
  </div>
</DocumentFragment>
`;
